### ISSUE TITLE
Content-Type of Response overwritten

### ISSUE THREAD

**Issue by topper-123** [encode/apistar#297]
The below app returns a json response:

```python
from apistar.frameworks.wsgi import WSGIApp as App
from apistar import Response, Route 

def welcome():
    data = '<h1>Welcome</h1>'
    headers = {'Content-Type': 'text/html; charset=utf-8'}
    return Response(data, headers=headers)

routes = [
    Route('/', 'GET', welcome),
]
app = App(routes=routes)

if __name__ == '__main__':
    app.main()
```

Expected (IMO) would be an html page. i.e. if ``content-type`` (or any other header) is set explicitly in the response instance, then this overrides default setttings...

**Comment by kinabalu** [comment]
the proper way to do this, is to pass `content_type` as an argument to Response, along with ensuring that the variable `data` is of type bytes:
```
data = b'<h1>Welcome</h1>'
```

This block should work for you:

```
from apistar.frameworks.wsgi import WSGIApp as App
from apistar import Response, Route

def welcome() -> Response:
    data = b'<h1>Welcome</h1>'
    headers = {'Content-Type': 'text/html; charset=utf-8'}
    return Response(content=data, headers=headers, content_type='text/html')

routes = [
    Route('/', 'GET', welcome),
]
app = App(routes=routes)

if __name__ == '__main__':
    app.main()
```

**Comment by martinpeck** [comment]
Personally, I find it better to annotate the handler, rather than explicitly set up headers and create Response objects within the handler...

```python
from apistar import annotate
from apistar.renderers import HTMLRenderer

@annotate(renderers=[HTMLRenderer()])
def welcome():
    data = '<h1>Welcome</h1>'
    return data

```

... especially if you're writing a number of these.

**Comment by tomchristie** [comment]
Agree with @martinpeck.

Tho I still think we have some counter-intuitive behavior in overwriting the content-type header like this, and probably still be worth addressing.

**Comment by tomchristie** [comment]
Resolved in #400. Also it's more explicit - either use `JSONResponse`, `HTMLResponse`, or instead just `Response` and specify the header explicitly.
